Tests/AblyChatTests/Mocks/MockRoomLifecycleManager.swift

```diff
@@ -3,28 +3,41 @@ import Ably
 
 actor MockRoomLifecycleManager: RoomLifecycleManager {
     private let attachResult: Result<Void, ARTErrorInfo>?
-    private(set) var attachCallCount = 0
     private let detachResult: Result<Void, ARTErrorInfo>?
+    private let resultOfWaitToBeAbleToPerformPresenceOperations: Result<Void, ARTErrorInfo>?
+
+    private(set) var attachCallCount = 0
     private(set) var detachCallCount = 0
     private(set) var releaseCallCount = 0
+    private(set) var waitCallCount = 0
+
     private let _roomStatus: RoomStatus?
     private var subscriptions = SubscriptionStorage<RoomStatusChange>()
 
-    init(attachResult: Result<Void, ARTErrorInfo>? = nil, detachResult: Result<Void, ARTErrorInfo>? = nil, roomStatus: RoomStatus? = nil) {
+    init(
+        attachResult: Result<Void, ARTErrorInfo>? = nil,
+        detachResult: Result<Void, ARTErrorInfo>? = nil,
+        roomStatus: RoomStatus? = nil,
+        resultOfWaitToBeAblePerformPresenceOperations: Result<Void, ARTErrorInfo>? = nil
+    ) {
         self.attachResult = attachResult
         self.detachResult = detachResult
+        resultOfWaitToBeAbleToPerformPresenceOperations = resultOfWaitToBeAblePerformPresenceOperations
         _roomStatus = roomStatus
     }
 
     func performAttachOperation() async throws(InternalError) {
         attachCallCount += 1
-        guard let attachResult else {
-            fatalError("In order to call performAttachOperation, attachResult must be passed to the initializer")
-        }
-        do {
-            try attachResult.get()
-        } catch {
-            throw error.toInternalError()
+        emitStatusChange(.init(current: .attaching(error: nil), previous: _roomStatus ?? .initialized))
```

Comment:
Based on my comments on the presence tests, I'm not convinced of the necessity of the changes to this method.

Reply:
reverted